### ISSUE TITLE
Decide on ui <-> client API

### ISSUE THREAD

**Issue by hkjn** [hkjn/lnhw#4]
I'm filing an issue to remind ourselves that we need to settle on an API between the `ui` layer and `client` layer.

On `ui` side, @superquest and friends have started with https://github.com/hkjn/lnhw/blob/master/ui/sdk.js to stub out the interactions they believe are needed for `ui` to fetch data from `client` to find out about the c-lightning state, as well as for `ui` to tell `client` to take an action, e.g. sending a BOLT11 invoice.

Some options:

1. Use `lightning-cli`, with a `ui`-side backend that receives HTTP requests from the client-side code, and translates them into `lightning-cli` commands
2. Build a small HTTP backend on `client` side, similar to the https://github.com/hkjn/lnmon/blob/master/lnmon.go tool which wraps `lightning-cli` to provide the data that is shown on the https://ln.hkjn.me web server + prometheus monitoring target

**Comment by superquest** [comment]
[This Python app](https://github.com/ElementsProject/lightning/tree/master/contrib/pylightning) might be a good basis for the http backend.

It's in the c-lightning source so that's probably a good sign ...

**Comment by hkjn** [comment]
Yeah, that would work. We'd still need some http REST -> .py call translation going on though, similar to what `lnmon.go` does for Go.. but if you make progress and have spare capacity in `ui` team, by all means you can get started on looking at what such a backend would look like.

**Comment by superquest** [comment]
Yes the lnmon looks like better fit.

**Comment by hkjn** [comment]
As discussed offline I was just suggesting looking at `lnmon.go`  to get an idea of the different `lightning-cli` commands that might be useful, and the data they output.. a small python app that served up a web server most likely would be a faster way to build the `ui` -> `client` API, in my opinion.

**Comment by hkjn** [comment]
Just to capture the latest here, we are currently noticing that we will need an additional field for the `lightning-cli getinfo` call to indicate whether the hardware `wallet` is connected or not for the `ui` layer.

We are suggesting something like:

```
{
  "id": "0264089bc3a4afde671b87740d72e03d899b7f0783fbb756786cb7632b2e1db510",
  "port": 9735,
  "address": [
    {
      "type": "ipv4",
      "address": "163.172.162.18",
      "port": 9735
    }
  ],
  "version": "v0.5.2-2016-11-21-2627-g78f3e8d8",
  "blockheight": 529898,
  "network": "bitcoin",
  "hardwarewallet": "connected"
}
```

If the `hardwarewallet` key is not present or not with the value `connected` in the `getinfo` output, we do not have a hardware wallet connected to the computer currently.

**Comment by hkjn** [comment]
Fyi we have a branch with support for `hardwarewallet` field in `getinfo` now: https://github.com/hkjn/lightning/tree/watch-bite1
